ckanext/saml2/model/user_sso_gen.py

```diff
@@ -0,0 +1,26 @@
+from sqlalchemy import Column, UnicodeText, Boolean
+from sqlalchemy.ext.declarative import declarative_base
+import ckan.model as model
+
+Base = declarative_base()
+metadata = Base.metadata
+
+
+class UserSsoGen(Base):
+    __tablename__ = 'saml2_user_sso_gen'
+
+    id = Column(UnicodeText, primary_key=True)
+    gen = Column(UnicodeText, nullable=False, unique=True)
```

Comment:
It will be better if the class, table and column naming (and anything else I've not spotted) doesn't contain 'gen' since a GEN is just a concept in NSW government not a general SAML2 term.  I suggest calling the table simply `saml2_user`, the column `name_id`, and the class `SAMLUser` or `SamlUser` (whichever case fits pep8)